compiler/rustc_mir_transform/src/elaborate_drops.rs

```diff
@@ -379,112 +377,38 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                                 Unwind::To(Option::unwrap_or(unwind, resume_block))
                             },
                             bb,
+                            is_replace,
                         ),
                         LookupResult::Parent(..) => {
-                            self.tcx.sess.delay_span_bug(
-                                terminator.source_info.span,
-                                &format!("drop of untracked value {:?}", bb),
+                            if !is_replace {
+                                self.tcx.sess.delay_span_bug(
+                                    terminator.source_info.span,
+                                    &format!("drop of untracked value {:?}", bb),
+                                );
+                            }
+                            // If we cannot find it, it means it's a drop followed by a replace
+                            // -> unconditional dro
+                            Elaborator { ctxt: self }.patch().patch_terminator(
+                                bb,
+                                TerminatorKind::Drop {
+                                    place,
+                                    target,
+                                    unwind: if data.is_cleanup {
+                                        None
+                                    } else {
+                                        Some(Option::unwrap_or(unwind, resume_block))
+                                    },
+                                    is_replace,
+                                },
```

Comment:
I don't follow this logic

Reply:
It's this case here https://github.com/rust-lang/rust/blob/270c94e484e19764a2832ef918c95224eb3f17c7/compiler/rustc_mir_transform/src/elaborate_drops.rs#L477, will add a better comment